### PR TITLE
Configure iptables to accept all TCP/UDP/ICMP packets.

### PR DESCRIPTION
**Pull request by Random-Liu**
Kubernetes node need to accept all TCP/UDP/ICMP packets. However, by default on travis trusty node, `FORWARD` is `DROP`. That's why all intra-pod communication tests are failing:
```
[91m[1m[Fail] [0m[90m[k8s.io] Networking [0m[0m[k8s.io] Granular Checks: Pods [0m[91m[1m[It] should function for intra-pod communication: udp [Conformance] [0m
[37m/home/travis/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:217[0m

[91m[1m[Fail] [0m[90m[k8s.io] Networking [0m[0m[k8s.io] Granular Checks: Pods [0m[91m[1m[It] should function for intra-pod communication: http [Conformance] [0m
[37m/home/travis/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/networking_utils.go:217[0m

[91m[1m[Fail] [0m[90m[k8s.io] Container Lifecycle Hook [0m[0mwhen create a pod with lifecycle hook [0m[91m[1m[It] should execute prestop exec hook properly [Conformance] [0m
[37m/home/travis/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/lifecycle_hook_test.go:83[0m

[91m[1m[Fail] [0m[90m[k8s.io] Container Lifecycle Hook [0m[0mwhen create a pod with lifecycle hook [0m[91m[1m[It] should execute poststart exec hook properly [Conformance] [0m
[37m/home/travis/gopath/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/lifecycle_hook_test.go:74[0m
```

The iptables rules are required for Kubernetes:
* Cluster setup: https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L83-L106
* Remote node e2e test: https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/remote/utils.go#L80-L112

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

**Comment by Random-Liu** [comment]
With this PR, we could pass all the intra-pod communication test now. Will close https://github.com/kubernetes-incubator/cri-containerd/pull/198 after this PR is merged.

File: hack/test-e2e-node.sh [hunk 1]
```diff
@@ -35,6 +35,19 @@ if [[ -z "${GOPATH}" ]]; then
   exit 1
 fi
 
+# Update ip firewall
+# We need to add rules to accept all TCP/UDP/ICMP packets.
+if sudo iptables -L INPUT | grep "Chain INPUT (policy DROP)" > /dev/null; then
+	sudo iptables -A INPUT -w -p TCP -j ACCEPT
```
Comment (mikebrow) [inline hack/test-e2e-node.sh]: Would this work better:
```sudo iptables -I INPUT 1 -w -j ACCEPT```
^ translates to insert at rule 1 in ACCEPT allow traffic for `all` protocols from anywhere to anywhere

File: hack/test-e2e-node.sh [hunk 2]
```diff
@@ -35,6 +35,19 @@ if [[ -z "${GOPATH}" ]]; then
...
+	sudo iptables -A INPUT -w -p UDP -j ACCEPT
+	sudo iptables -A INPUT -w -p ICMP -j ACCEPT
+fi
+if sudo iptables -L FORWARD | grep "Chain FORWARD (policy DROP)" > /dev/null; then
+	sudo iptables -A FORWARD -w -p TCP -j ACCEPT
```
Comment (mikebrow) [inline hack/test-e2e-node.sh]: ```sudo iptables -I FORWARD 1 -w -j ACCEPT```

File: hack/test-e2e-node.sh (hunk 1 above)
Comment (Random-Liu) [reply hack/test-e2e-node.sh]: Discussed with our networking team. I believe we want to keep the existing filter, and accept at the end.

File: hack/test-e2e-node.sh (hunk 2 above)
Comment (Random-Liu) [reply hack/test-e2e-node.sh]: Same.

**Comment by Random-Liu** [comment]
@mikebrow Added the `iptables-save` and `iptables-restore` logic. Really good suggestion! We should try not to mess up developer's node... :p

**Comment by Random-Liu** [comment]
Will merge after test passes.

**Comment by Random-Liu** [comment]
@mikebrow I'll not merge this even test passes. Yours is the next one to merge! ;)

**Comment by mikebrow** [comment]
lolz I don't mind, it's all good :-)

**Pull request merged**